### PR TITLE
Modify IndexFactory.createTabixIndex() for bgzip input

### PR DESCRIPTION
**Pull request by abaumann**
Modify the internal FeatureIterator.initStream method inside
IndexFactory to accept both block-compressed and non-compressed
input files. This method leaves the constructor as is and does
the work without extra work for the user.  The method will
check that this is proper bgzip data, and throw exception if
not.  The unit test checks that the same result comes from calling
this method with a VCF and the same bgzipped VCF.

Resolves: #184

### CODE REVIEW

**Comment by nh13** [comment]
@jmthibault79 or @mattsooknah or @droazen, can you take a look at this?

File: src/java/htsjdk/tribble/index/IndexFactory.java [hunk 1]
```diff
@@ -420,8 +419,32 @@ private FeatureCodecHeader readHeader() {
...
+                    final int bufferSize = Math.max(Defaults.BUFFER_SIZE, BlockCompressedStreamConstants.MAX_COMPRESSED_BLOCK_SIZE);
+                    final BufferedInputStream bufferedStream = new BufferedInputStream(fileStream, bufferSize);
+
+                    if (!BlockCompressedInputStream.isValidFile(bufferedStream)) {
+                        throw new TribbleException.MalformedFeatureFile("Input file is not in valid block compressed format.", inputFile.getAbsolutePath());
+                    }
+
+                    // if we got here, the file is valid, make a SeekableStream for the BlockCompressedInputStream to read from
+                    final SeekableStream seekableStream =
+                            SeekableStreamFactory.getInstance().getBufferedStream(
```
Comment (jmthibault79) [inline src/java/htsjdk/tribble/index/IndexFactory.java]: I would save a local reference to the `SeekableStreamFactory` instance here, to make it a little more readable.

But that's just personal preference, ignore if you disagree.

**Comment by jmthibault79** [comment]
I like this approach.  :+1: from me.

**Comment by abaumann** [comment]
slight modification to reuse SeekableStreamFactory.getInstance(), if build passes, shall I merge?

**Comment by jmthibault79** [comment]
Yes go ahead.

**Pull request merged**